code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -84,6 +84,47 @@
 		if(standard_pour_into(user, target))
 			return
 
+		//Help the user avoid clumsy splashing mistakes due to finicky sprites etc.
+		if (reagents.total_volume) //don't pop a dialog if it's empty...
+			//handle lidded containers
+			if  (user.clumsy == 0 && (istype(target, /obj/item/weapon/reagent_containers)))
+				var/resp = alert(user, "Do you really want to splash the solution?", "Splash the solution?", "Abort", "Proceed")
+				if (resp == "Abort")
+					return
+				else if (!user.Adjacent(target))
+					user << "You cannot reach the [src]."
+					return
+				else if (user.stat != 0)
+					return
+				else if (src.loc != user)
+					user << "You need to hold the [src] to splash with it."
+					return
+
+			//handles splashes onto floors and effects (puddles, blood etc.)
+			if (user.clumsy <= 1 && (istype(target, /turf) || istype(target, /obj/effect) ))
+
+				//resolve contentless effects
+				var/intarget = null
+				if (istype(target, /obj/effect))
+					intarget = target.loc
+				else intarget = target
+
+				//iterate over things in content
+				for (var/thing in intarget)
+					if (is_type_in_list(thing, can_be_placed_into))
+						var/response = alert(user, "Do you really want to splash the solution?", "Splash the solution?", "Abort", "Proceed")
+						if (response == "Abort")
+							return
+						else if (!user.Adjacent(target))
+							user << "You cannot reach that."
+							return
+						else if (user.stat != 0) //mob isn't fully alive
+							return
+						else if (src.loc != user) //mob doesn't carry the container in an equipment slot
+							user << "You need to hold the [src] to splash with it."
+							return
+					break
```

Comment:
this is only going to hit the "first" item in the contents, which is the first item added to it, which due to BYOND being BYOND can be random as fuck, this is a horrible way to go about this